source/module_beyonddft/dm_trans/test/dm_trans_test.cpp

```diff
@@ -0,0 +1,277 @@
+#include "../dm_trans.h"
+
+#include "mpi.h"
+
+#include <gtest/gtest.h>
+#ifdef __MPI
+#include "module_beyonddft/utils/lr_util.h"
+#endif
+struct matsize
+{
+    int nks = 1;
+    int naos;
+    int nocc;
+    int nvirt;
+    int nb = 1;
+    matsize(int nks, int naos, int nocc, int nvirt, int nb = 1) : nks(nks), naos(naos), nocc(nocc), nvirt(nvirt), nb(nb)
+    {
+        assert(nocc + nvirt <= naos);
+    };
+};
+
+class DMTransTest : public testing::Test
+{
+  public:
+    std::vector<matsize> sizes{{2, 14, 9, 4}, {2, 20, 10, 7}};
+    int nstate = 2;
+    std::ofstream ofs_running;
+    int my_rank;
```

Comment:
It is better to give my_rank, naos, nocc, nvirt initial values.

Reply:
OK, done.